### PR TITLE
Restore DOMOption warnings for Object and Fragment children (#13586)

### PR DESCRIPTION
**Pull request by Simek**
This pull request restores missing warning for invalid `option` tag children (Object and Fragment) mentioned by @gaearon in #13586.

Two tests has been added according to examples provided in issue.

Any suggestions are welcome and appreciated.

### CODE REVIEW

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 12f3a5475fbd07d4f0806c5adfbb2b66c264a643...e1f0d92772acb1459becd96ee643f3e405e8fccd</p>


  
## schedule
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
schedule.development.js | n/a | n/a | 0 B | 19.17 KB | 0 B | 5.74 KB | UMD_DEV
**schedule.production.min.js** | **n/a** | **n/a** | **0 B** | **3.16 KB** | **0 B** | **1.53 KB** | **UMD_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/react-dom/src/client/ReactDOMOption.js [hunk 1]
```diff
@@ -53,7 +54,11 @@ export function validateProps(element: Element, props: Object) {
         if (typeof child === 'string' || typeof child === 'number') {
           return;
         }
-        if (typeof child.type !== 'string') {
```
Comment (gaearon) [inline packages/react-dom/src/client/ReactDOMOption.js]: Can this just be `=== 'function'`?

Comment (Simek) [reply packages/react-dom/src/client/ReactDOMOption.js]: Sure, it's so obvious now, thank you!

**Comment by Simek** [comment]
@gaearon Friendly ping after three months.

**Comment by Simek** [comment]
Issue #13586 has been closed without explanation. How does this decision apply to this PR?

**Comment by gaearon** [comment]
Sorry this slipped through the cracks. We fixed this in 18 alpha.
https://codesandbox.io/s/thirsty-austin-yzgyg?file=/src/index.js

**Pull request closed without merging**